ctapipe/reco/shower_processor.py

```diff
@@ -0,0 +1,193 @@
+"""
+High level processing of showers.
+
+This processor will be able to process a shower/event in 3 steps:
+- shower geometry
+- estimation of energy (optional, currently unavailable)
+- estimation of classification (optional, currently unavailable)
+
+"""
+import numpy as np
+
+from astropy.coordinates import SkyCoord, AltAz
+
+from ctapipe.core import Component, QualityQuery
+from ctapipe.core.traits import List
+from ctapipe.containers import ArrayEventContainer, ReconstructedShowerContainer
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.reco import HillasReconstructor
+
+
+DEFAULT_SHOWER_PARAMETERS = ReconstructedShowerContainer()
+
+
+class ShowerQualityQuery(QualityQuery):
+    """Configuring shower-wise data checks."""
+
+    quality_criteria = List(
+        default_value=[
+            ("Finite charge", "lambda hillas: hillas.intensity > 0"),
+            ("Finite width", "lambda hillas: hillas.width > 0"),
+            ("Finite length", "lambda hillas: hillas.length > 0"),
+            ("Ellipticity > 0.1", "lambda hillas: hillas.width / hillas.length > 0.1"),
```

Comment:
How were these cuts here determined? Are they good enough as defaults?

Reply:
The first 3 are minimum conditions I would say, the ellipticity one is the one used in the protopipe/CTAMARS analysis.
These are just defaults, so we can change them if needed.

the defaults of image processor are size >50 and n_pixels> 3 so the first 3 cuts are actually overwritten now that I think about it.

Reply:
They are needed to ignore non-reconstructed telescope events.

Reply:
if we want the defaults to just allow reconstruction, then the ellipticity cut is probably too much

I would at least make sure that width and length are positive, but in case someone changes the defaults of image processor for any reason I would keep also the intensity one

maybe we could add n_pixels also here

the containment cut (or leakage) is probably more related to performance like the ellipticity

Reply:
These checks are completely independent of the `ImageQualityQuery`, since all events will be processed here. So we need to do all queries we want to do and we cannot rely on a pre-selection by `ImageQualityQuery`.

Reply:
OK I would say then to make it coherent with ImageProcessor, so

- intensity > 50 phe 
- n_pixels > 3

this should automatically ensure that both width and length are positive, though it is possible (but very rare) that width can be 0, so I would also add that as a check.

Reply:
Though to access n_pixels I have to start directly from DL1EventSource instead of the hillas dictionary

Reply:
done